from PyQt5.QtWidgets import QApplication, QMainWindow, QMessageBox
from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.Qt import QIcon, QFont
from controller.VendaCTR import VendaCTR

class Ui_FormLancarCredito(object):


    def preencher_campos(self, codvenda, valorvenda, valorpago, cliente):
        self.valorvenda = float(valorvenda)
        self.valorpago = float(valorpago)
        Valorvenda = ("R$ {}".format(valorvenda))
        Valorpago = ("R$ {}".format(valorpago))
        self.editCodVenda.setText(str(codvenda))
        self.editValorVenda.setText(Valorvenda)
        self.editValorPago.setText(Valorpago)
        self.editCliente.setText(str(cliente))

    def click_btn_lancar(self):
        codvenda = int(self.editCodVenda.text())
        valorvenda = self.valorvenda
        valorpago = self.valorpago
        credito = self.creditoSpinBox.value()

        if ((credito+valorpago) > valorvenda):
            msg = QMessageBox(None)
            msg.setWindowIcon(QIcon("error.png"))
            msg.setWindowTitle("Sucesso!")
            msg.setIcon(QMessageBox.Critical)
            font = QFont()
            font.setFamily("Arial")
            font.setPointSize(15)
            msg.setFont(font)
            text = ("Erro: O valor do crédito não pode ser maior que o valor da venda.")
            msg.setText(text)
            msg.exec_()

        elif (credito == 0):
            msg = QMessageBox(None)
            msg.setWindowIcon(QIcon("error.png"))
            msg.setWindowTitle("Erro")
            msg.setIcon(QMessageBox.Critical)
            font = QFont()
            font.setFamily("Arial")
            font.setPointSize(15)
            msg.setFont(font)
            text = ("Erro: O valor do crédito não pode ser igual a 0 (zero).")
            msg.setText(text)
            msg.exec_()

        elif ((credito+valorpago) <= valorvenda):
            vlvenda = self.valorvenda
            vlpago = self.valorpago
            Credito = credito
            vendaCTR = VendaCTR
            vendaCTR.lancar_credito(codvenda, vlvenda, vlpago, Credito)



    def setupUi(self, FormLancarCredito):
        FormLancarCredito.setObjectName("FormLancarCredito")
        FormLancarCredito.resize(499, 326)
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("credit.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        FormLancarCredito.setWindowIcon(icon)
        self.centralwidget = QtWidgets.QWidget(FormLancarCredito)
        self.centralwidget.setObjectName("centralwidget")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(190, 20, 121, 23))
        font = QtGui.QFont()
        font.setPointSize(14)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.labelCredito = QtWidgets.QLabel(self.centralwidget)
        self.labelCredito.setGeometry(QtCore.QRect(90, 190, 51, 16))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.labelCredito.setFont(font)
        self.labelCredito.setObjectName("labelCredito")
        self.creditoSpinBox = QtWidgets.QDoubleSpinBox(self.centralwidget)
        self.creditoSpinBox.setGeometry(QtCore.QRect(140, 190, 81, 20))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.creditoSpinBox.setFont(font)
        self.creditoSpinBox.setPrefix("")
        self.creditoSpinBox.setSuffix("")
        self.creditoSpinBox.setMaximum(999.99)
        self.creditoSpinBox.setObjectName("creditoSpinBox")
        self.labelCodVenda = QtWidgets.QLabel(self.centralwidget)
        self.labelCodVenda.setGeometry(QtCore.QRect(50, 70, 81, 16))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.labelCodVenda.setFont(font)
        self.labelCodVenda.setObjectName("labelCodVenda")
        self.layoutWidget = QtWidgets.QWidget(self.centralwidget)
        self.layoutWidget.setGeometry(QtCore.QRect(170, 250, 151, 41))
        self.layoutWidget.setObjectName("layoutWidget")
        self.horizontalLayout = QtWidgets.QHBoxLayout(self.layoutWidget)
        self.horizontalLayout.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.btnLancar = QtWidgets.QPushButton(self.layoutWidget)
        icon1 = QtGui.QIcon()
        icon1.addPixmap(QtGui.QPixmap("credit.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.btnLancar.setIcon(icon1)
        self.btnLancar.setIconSize(QtCore.QSize(25, 25))
        self.btnLancar.setObjectName("btnLancar")
        self.horizontalLayout.addWidget(self.btnLancar)
        self.btnLimpar = QtWidgets.QPushButton(self.layoutWidget)
        icon2 = QtGui.QIcon()
        icon2.addPixmap(QtGui.QPixmap("borracha.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.btnLimpar.setIcon(icon2)
        self.btnLimpar.setIconSize(QtCore.QSize(25, 25))
        self.btnLimpar.setObjectName("btnLimpar")
        self.horizontalLayout.addWidget(self.btnLimpar)
        self.editCodVenda = QtWidgets.QLineEdit(self.centralwidget)
        self.editCodVenda.setGeometry(QtCore.QRect(140, 70, 61, 20))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.editCodVenda.setFont(font)
        self.editCodVenda.setInputMask("")
        self.editCodVenda.setText("")
        self.editCodVenda.setEchoMode(QtWidgets.QLineEdit.Normal)
        self.editCodVenda.setReadOnly(True)
        self.editCodVenda.setObjectName("editCodVenda")
        self.editCliente = QtWidgets.QLineEdit(self.centralwidget)
        self.editCliente.setGeometry(QtCore.QRect(140, 160, 231, 20))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.editCliente.setFont(font)
        self.editCliente.setInputMask("")
        self.editCliente.setText("")
        self.editCliente.setEchoMode(QtWidgets.QLineEdit.Normal)
        self.editCliente.setReadOnly(True)
        self.editCliente.setObjectName("editCliente")
        self.labelCliente = QtWidgets.QLabel(self.centralwidget)
        self.labelCliente.setGeometry(QtCore.QRect(90, 160, 51, 16))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.labelCliente.setFont(font)
        self.labelCliente.setObjectName("labelCliente")
        self.editValorVenda = QtWidgets.QLineEdit(self.centralwidget)
        self.editValorVenda.setGeometry(QtCore.QRect(140, 100, 81, 20))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.editValorVenda.setFont(font)
        self.editValorVenda.setInputMask("")
        self.editValorVenda.setText("")
        self.editValorVenda.setEchoMode(QtWidgets.QLineEdit.Normal)
        self.editValorVenda.setReadOnly(True)
        self.editValorVenda.setObjectName("editValorVenda")
        self.labelValorVenda = QtWidgets.QLabel(self.centralwidget)
        self.labelValorVenda.setGeometry(QtCore.QRect(30, 100, 111, 20))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.labelValorVenda.setFont(font)
        self.labelValorVenda.setObjectName("labelValorVenda")
        self.labelValorPago = QtWidgets.QLabel(self.centralwidget)
        self.labelValorPago.setGeometry(QtCore.QRect(60, 130, 81, 20))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.labelValorPago.setFont(font)
        self.labelValorPago.setObjectName("labelValorPago")
        self.editValorPago = QtWidgets.QLineEdit(self.centralwidget)
        self.editValorPago.setGeometry(QtCore.QRect(140, 130, 81, 20))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.editValorPago.setFont(font)
        self.editValorPago.setInputMask("")
        self.editValorPago.setText("")
        self.editValorPago.setEchoMode(QtWidgets.QLineEdit.Normal)
        self.editValorPago.setReadOnly(True)
        self.editValorPago.setObjectName("editValorPago")
        FormLancarCredito.setCentralWidget(self.centralwidget)
        self.statusbar = QtWidgets.QStatusBar(FormLancarCredito)
        self.statusbar.setObjectName("statusbar")
        FormLancarCredito.setStatusBar(self.statusbar)

        self.retranslateUi(FormLancarCredito)
        self.btnLimpar.clicked.connect(self.editCodVenda.clear)
        self.btnLimpar.clicked.connect(self.creditoSpinBox.clear)
        QtCore.QMetaObject.connectSlotsByName(FormLancarCredito)
        FormLancarCredito.setTabOrder(self.editCodVenda, self.creditoSpinBox)
        FormLancarCredito.setTabOrder(self.creditoSpinBox, self.btnLancar)
        FormLancarCredito.setTabOrder(self.btnLancar, self.btnLimpar)

        self.btnLancar.clicked.connect(self.click_btn_lancar)

    def retranslateUi(self, FormLancarCredito):
        _translate = QtCore.QCoreApplication.translate
        FormLancarCredito.setWindowTitle(_translate("FormLancarCredito", "Lançar Credito"))
        self.label.setText(_translate("FormLancarCredito", "Lançar Crédito"))
        self.labelCredito.setText(_translate("FormLancarCredito", "Crédito:"))
        self.labelCodVenda.setText(_translate("FormLancarCredito", "Cód. Venda:"))
        self.btnLancar.setText(_translate("FormLancarCredito", "Lançar"))
        self.btnLimpar.setText(_translate("FormLancarCredito", "Limpar"))
        self.labelCliente.setText(_translate("FormLancarCredito", "Cliente:"))
        self.labelValorVenda.setText(_translate("FormLancarCredito", "Valor da Venda:"))
        self.labelValorPago.setText(_translate("FormLancarCredito", "Valor Pago:"))

if __name__ == '__main__':
    import sys
    app = QApplication(sys.argv)
    teste = QMainWindow()
    ui = Ui_FormLancarCredito()
    ui.setupUi(teste)
    teste.show()
    sys.exit(app.exec_())